### PR TITLE
Tests, CI, and CMake

### PR DESCRIPTION
**Pull request by kgerheiser**
Re-factor CMake

Automatically build and test on push and commits

Remove version number from lib name

Tests pass here: 

https://github.com/kgerheiser/NCEPLIBS-nemsio/actions/runs/128827631

### CODE REVIEW

**Comment by kgerheiser** [comment]
Hmm, I accidentally deleted the old build system. I wanted to keep that in the develop branch.

**Comment by kgerheiser** [comment]
Everything is fixed

File: utils/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+add_executable(nemsio_read nemsio_read.f90)
+target_link_libraries(nemsio_read nemsio bacio::bacio_4 w3nco::w3nco_d)
```
Comment (aerorahul) [inline utils/CMakeLists.txt:2]: on second thought, `bacio` and `w3nco` are not used in these utilities. Don't need them here.

**Comment by kgerheiser** [comment]
Just linking to bacio and wn3co, or something else?

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+set(srcName read_nemsio.f90)
+set(exeName read_nemsio.x)
+
+execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink
+                ${CMAKE_CURRENT_SOURCE_DIR}/testdataset
+                ${CMAKE_CURRENT_BINARY_DIR}/testdataset)
+
+add_executable(${exeName} ${srcName})
+add_dependencies(${exeName} nemsio bacio::bacio_4 w3nco::w3nco_d)
+target_link_libraries( ${exeName} PUBLIC nemsio bacio::bacio_4 w3nco::w3nco_d)
```
Comment (aerorahul) [inline tests/CMakeLists.txt:12]: `bacio` and `w3nco` are not used in this executable too. Please remove.

**Comment by kgerheiser** [comment]
Fixed

File: utils/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+add_executable(nemsio_get nemsio_get.f90)
+target_link_libraries(nemsio_get nemsio)
+
+add_executable(nemsio_chgdate nemsio_chgdate.f90)
+target_link_libraries(nemsio_chgdate nemsio)
+
+add_executable(mkgfsnemsioctl mkgfsnemsioctl.f90)
+target_link_libraries(mkgfsnemsioctl nemsio)
+
+install(TARGETS nemsio_read    RUNTIME DESTINATION bin)
```
Comment (aerorahul) [inline utils/CMakeLists.txt:13]: If you haven't merged in yet, may be you could add the `COMPONENT` tag here:
```
install(TARGETS nemsio_read RUNTIME DESTINATION bin COMPONENT utilities)
```

**Comment by kgerheiser** [comment]
Done

**Pull request merged**